indexed_gzip/zran.c

```diff
@@ -1662,24 +1670,48 @@ static int _zran_inflate(zran_index_t *index,
              * size is an exact multiple of
              # the read buffer size.
              */
-            if ((uint64_t) ftell_(index->fd, index->f) >= index->compressed_size &&
-                strm->avail_in <= 8) {
-
-                zran_log("End of file, stopping inflation\n");
+            if ((uint64_t) ftell_(index->fd, index->f) >= index->compressed_size) {
+                if (strm->avail_in >= 9) {
+                    /* We have two cases here: A) everything remaining in strm->next_in is null bytes
```

Comment:
I hope this makes sense @pauldmccarthy . I had to draw a diagram to get it all straight. Let me know if this is still unclear and I can try to add additional explanation.